lib/patterns.rb

```diff
@@ -12,4 +12,5 @@ module Patterns
   URL_VALIDATION_REGEXP = %r{\Ahttps?://([^\s:@]+:[^\s:@]*@)?[A-Za-z\d-]+(\.[A-Za-z\d-]+)+\.?(:\d{1,5})?([/?]\S*)?\z}
   VERSION_PATTERN       = /\A#{Gem::Version::VERSION_PATTERN}\z/o
   REQUIREMENT_PATTERN   = Gem::Requirement::PATTERN
+  BASE64_PATTERN        = %r{\A(?:[a-zA-Z0-9+/]{4})*(?:|(?:[a-zA-Z0-9+/]{3}=)|(?:[a-zA-Z0-9+/]{2}==)|(?:[a-zA-Z0-9+/]{1}===))\z}
```

Comment:
This is the base64 of a sha256 hash, right? Should we use a general base64 pattern for a known length hash?

I used the following in Bundler but it would be slightly different (`===` never happens, as far as I know, or I made a mistake in Bundler):

```
%r{\A[-0-9a-z_+/]{43}={0,2}\z}i
```